lib/pbench/server/cache_manager.py

```diff
@@ -465,15 +492,13 @@ def _get_metadata(tarball_path: Path) -> Optional[JSONOBJECT]:
         name = Dataset.stem(tarball_path)
         try:
             data = Tarball.extract(tarball_path, f"{name}/metadata.log")
-        except TarballUnpackError:
-            data = None
-        if data:
-            metadata = MetadataLog()
-            metadata.read_string(data)
-            metadata = {s: dict(metadata.items(s)) for s in metadata.sections()}
-            return metadata
-        else:
+        except BadDirpath:
             return None
+        else:
+            metadata_log = MetadataLog()
+            metadata_log.read_file(data)
```

Comment:
`Tarball.extract()` (now) returns a _binary_ stream, but `MetadataLog.read_file()` want a _text_ stream, so we have to convert:
```python
            metadata_log.read_file(e.decode() for e in data)
```